Formula/httpd.rb

```diff
@@ -18,14 +18,188 @@ class Httpd < Formula
   depends_on "pcre"
 
   def install
-    # use Slackware-FHS layout as it's closest to what we want.
-    # these values cannot be passed directly to configure, unfortunately.
-    inreplace "config.layout" do |s|
-      s.gsub! "${datadir}/cgi-bin", "#{pkgshare}/cgi-bin"
-      s.gsub! "${datadir}/error",   "#{pkgshare}/error"
-      s.gsub! "${datadir}/htdocs",  "#{pkgshare}/htdocs"
-      s.gsub! "${datadir}/icons",   "#{pkgshare}/icons"
-    end
+    # point config files to opt_prefix instead of the version-specific prefix
+    inreplace "Makefile.in",
+      '#@@ServerRoot@@#$(prefix)#', '#@@ServerRoot@@'"##{opt_prefix}#"
+
+    # fix default user/group when running as root
+    inreplace "docs/conf/httpd.conf.in", /(User|Group) daemon/, "\\1 _www"
+
+    Pathname.new("docs/conf/httpd.conf.in").append_lines <<-EOS.undent
+      # Welcome page
+      Include #{etc}/httpd/extra/httpd-welcome.conf
+
+      # Additional configuration
+      IncludeOptional #{etc}/httpd/other/*.conf
+    EOS
+
+    Pathname.new("docs/conf/extra/httpd-welcome.conf").write <<-EOS.undent
+      # This enables the default Welcome page if there
+      # is no default index page present for the root URL.
+      <LocationMatch "^/+$">
+          Options -Indexes
+          ErrorDocument 403 /error/noindex.html
+      </LocationMatch>
+
+      <Directory "#{var}/www/error">
+          AllowOverride None
+          Require all granted
+      </Directory>
+      <Directory "#{var}/www/icons">
+          Options Indexes MultiViews
+          AllowOverride None
+          Require all granted
+      </Directory>
+
+      Alias /error/noindex.html "#{var}/www/error/noindex.html"
+      Alias /icons/ "#{var}/www/icons/"
+    EOS
+
+    Pathname.new("docs/error/noindex.html").write <<-EOS.undent
```

Comment:
Sorry, definitely don't want to include this stuff. The upstream defaults are fine.

Reply:
That's a bummer, although I did put it in a separate commit in such a case.

The upsteam deafults don't really provide a good starting point for people that aren't familiar with httpd, it's configuration paths, and what all it installs (especially for folks that get it installed as a dependency). Having this extra error document with welcome information is _way_ more helpful than the default index that is installed that just says, "It works!". Can't we provide a little helping hand for our just getting started devs? It also provides some branding that reenforces the Homebrew project.

Reply:
> Having this extra error document with welcome information is way more helpful than the default index that is installed that just says, "It works!". Can't we provide a little helping hand for our just getting started devs? It also provides some branding that reenforces the Homebrew project.

Sorry, no to this.

Reply:
🤷‍♂️ don't hate a dev for trying. 🙂